backend/ohq/models.py

```diff
@@ -447,3 +449,180 @@ class Meta:
 
     def __str__(self):
         return f"{self.user}: {self.metric}"
+    
+class Occurrence(models.Model):
+    event = models.ForeignKey(Event, on_delete=models.CASCADE, verbose_name=("event"), related_name="occurrences")
+    title = models.CharField(("title"), max_length=255, blank=True)
+    location = models.CharField(("location"), max_length=255, blank=True)
+    description = models.TextField(("description"), blank=True)
+    start = models.DateTimeField(("start"), db_index=True)
+    end = models.DateTimeField(("end"), db_index=True)
+    cancelled = models.BooleanField(("cancelled"), default=False)
+    original_start = models.DateTimeField(("original start"))
+    original_end = models.DateTimeField(("original end"))
+    created_on = models.DateTimeField(("created on"), auto_now_add=True)
+    updated_on = models.DateTimeField(("updated on"), auto_now=True)
+    interval = models.IntegerField(("interval"), blank=True, null=True)
+
+    class Meta:
+        verbose_name = ("occurrence")
+        verbose_name_plural = ("occurrences")
+        index_together = (("start", "end"),)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        event = kwargs.get("event", None)
+        if not self.title and event:
+            self.title = event.title
+        if not self.description and event:
+            self.description = event.description
+        if not self.location and event:
+            self.location = event.location
+
+    def save(self, *args, **kwargs):
+        super().save(*args, **kwargs)
+
+        if self.interval is not None:
+            if self.pk: # If save is called on object update, not creation
+                self.bookings.all().delete()
+
+            delta = self.end - self.start
+            delta_minutes = delta.total_seconds() / 60
+            booking_count = int(delta_minutes // self.interval)
```

Comment:
I'm not a big fan of making booking objects automatically on save. I'm not totally against it since I see the value in just having the objects be there, but in some sense we are creating multiple sources of truths (obviously bad!!) --> i.e. we require the bookings to be synchronized with the interval field, hence why we are overriding save to make sure of this but this is generally kind of sus.

Reply:
For example maybe one way to avoid this is to initially have no bookings and allow users to create bookings, but you validate the time they specified (valid times can be derived from the start time and interval fields)

Reply:
Also not sure how events fit into the picture

Reply:
Lets talk next time I'm at gbm